src/UIState/NumberCollectorState.h

```diff
@@ -0,0 +1,61 @@
+/**
+ * Abstract superclass for the NumberCollector UI states
+ */
+
+#pragma once
+#include "UIState.h"
+
+class NumCollectorState : public UIState {
+public:
+  NumCollectorState() {
+    LiquidCrystal_TC::instance()->writeLine(" ", 1);
+  }
+  void handleKey(char key);
+  virtual void setValue(double value) = 0;
+  virtual int getExpectedDigits() = 0;
+
+protected:
+  // Helper Functions
+  void handleDigit(double digit);
+  void backSpace();
+  void printValue();
+  bool collectedEnoughDigits() {
+    return numDigits >= getExpectedDigits();
+  }
+
+  double value = 0.0;
+  int numDigits = 0;
+  int factor = 10;
+  bool noDecimal = true;
```

Comment:
You might think about phrasing booleans in a positive manner. So `hasDecimal` is `true` or `false` instead of `no` being `true`.